boot/mbed/app_enc_keys.c

```diff
@@ -30,8 +30,6 @@ extern unsigned int ecdsa_pub_key_len;
 #define HAVE_KEYS
 extern const unsigned char ed25519_pub_key[];
 extern unsigned int ed25519_pub_key_len;
-#else
-#error "No public key available for given signing algorithm."
```

Comment:
Is this file supposed to be excluded from compilation for no signature configuration?

Reply:
As far as I know, Mbed CLI1 cannot conditionally ignore files. Assuming files will be compiled and disabling unused code with macros is the only way.

Reply:
The linker also might leave out symbols that aren't used. It might not be necessary to conditionalize the entire file.